src/main/java/com/google/devtools/build/lib/analysis/constraints/RuleContextConstraintSemantics.java

```diff
@@ -908,7 +908,7 @@ public static ConfiguredTarget incompatibleConfiguredTarget(
       OrderedSetMultimap<DependencyKind, ConfiguredTargetAndData> prerequisiteMap)
       throws ActionConflictException, InterruptedException {
     // The target (ruleContext) is incompatible if explicitly specified to be.
-    if (ruleContext.getRule().getRuleClassObject().useToolchainResolution()
+    if (ruleContext.getRule().getRuleClass() != "toolchain"
```

Comment:
WDYT about this, @katre ? It's not very elegant, but I can't find anything cleaner that expresses the same thing.

Reply:
I think this makes sense, but it'd be good to add a comment with the general definition of `target_compatible_with` that any rule that needs its own meaning for `target_compatible_with` needs to be added to this list.

Also, make sure you're checking String equality with `String.equals`, not `==`.

Reply:
Dang. Every time I feel like I'm getting the hang of Java there's something else that trips me up :P

I fixed the string comparison. Thanks!

By "a comment with the general definition of `target_compatible_with`" do you mean updating `attributes/common/target_compatible_with.html`?  For now I added a comment for this right here in the code.

Reply:
Sorry, I meant adding a comment to BaseRuleClasses.NativeBuildRule.

Reply:
Gotcha, makes sense. Thanks!

Done